code/_helpers/icons.dm

```diff
@@ -927,3 +927,10 @@ proc/generate_image(var/tx as num, var/ty as num, var/tz as num, var/range as nu
 
 	return cap
 
+/atom/proc/add_overlay(var/overlay)
```

Comment:
Why though? Just use native overlay manipulation.

Reply:
Author posted something about this in PR description:
> Lets add_overlay use images instead of just icon_states, to let add_overlay work with icon files that aren't the src object's icon.